Source/GameView/Repositories/GameAPIRepo.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  GameAPIRepo.swift
+//  CommonFiles
+//
+//  Created by Tiewhan Smith on 2020/03/06.
+//
+
+import Foundation
+
+///Structure is only used to model JSON respones. Contains the a structure of Apps
+struct AppList: Codable {
+  let applist: Apps
+}
+
+///Structure is used to model JSON respones. Contains the list of games
+struct Apps: Codable {
+  let apps: [JsonGame]
+}
+
+///Structure is used to model JSON respones. This is the actual game
+struct JsonGame: Codable {
+  let appid: Int
+  let name: String
+}
+
+public class GameAPIRepo: GameRepository {
+  
+  private var observers: [String: GameRepositoryObserver] = [:]
+  
+  public init() {
+    
+  }
+  
+  public func getGameList(with serviceCaller: ServiceCaller) {
+
+    let url: URL = URL(string: SteamURLComponents.gameListURL)!
+    
+    // MARK: - Call Failed Closure
+    serviceCaller.callFailed = { error in
+      
+    }
+    
+    // MARK: - ---------------------------
+
+    // MARK: - Call Succeeded Closure
+    serviceCaller.callSucceeded = { [weak self] data, _ in
+
+      guard let self = self else {
+        return
+      }
+
+      let games = self.decodeGames(data: data)
+
+      var smallGameList: [Game] = []
+
+      //This limit exists until sectioning with the table works.
+      let size: Int = games.count >= 25 ? 25 : games.count
+
+      if games.count >= 1 {
+
+        for index in 0...(size - 1) {
+          smallGameList.append(games[index])
+        }
+
+        self.notifyGameListLoaded(gameList: smallGameList)
+
+      }
+
+    }
+    
+    // MARK: - ---------------------------
+    
+    do {
+      try serviceCaller.makeServiceCall(with: url, and: DataBundle())
```

Comment:
I like the usage of `with` but the `and` might be a bit problematic as someone using this function might not know they need to pass a DataBundle. If the parameters that need to be passed aren't obvious there's nothing wrong with the param being `dataBundle` or even `andBundle`

Reply:
I didn't think about that but that makes sense and greatly appreciate the input, eh